mlir-tensorrt/tensorrt/include/mlir-tensorrt-dialect/Utils/Types.h

```diff
@@ -0,0 +1,12 @@
+#ifndef MLIR_TENSORRT_UTILS_TYPES_H
+#define MLIR_TENSORRT_UTILS_TYPES_H
+
+#include "mlir-c/IR.h"
+#include <functional>
+#include <string>
+
+namespace mlirtrt {
```

Comment:
We go through a lot of trouble to have a C-pure interface for a reason. You can't just mix in C++.